plugins/http_plugin/http_plugin.cpp

```diff
@@ -275,17 +329,41 @@ namespace eosio {
             valid_hosts.emplace(host + ":" + resolved_port_str);
          }
 
+         void mangle_option_names() {
+            if(current_http_plugin_defaults.address_config_prefix.empty())
+               return;
+            unix_socket_path_option_name.insert(0, current_http_plugin_defaults.address_config_prefix+"-");
+            http_server_address_option_name.insert(0, current_http_plugin_defaults.address_config_prefix+"-");
+            https_server_address_option_name.insert(0, current_http_plugin_defaults.address_config_prefix+"-");
+         }
    };
 
+   template<>
+   bool http_plugin_impl::allow_host<detail::asio_local_with_stub_log>(const detail::asio_local_with_stub_log::request_type& req, websocketpp::server<detail::asio_local_with_stub_log>::connection_ptr con) {
+      return true;
+   }
+
    http_plugin::http_plugin():my(new http_plugin_impl()){}
    http_plugin::~http_plugin(){}
 
    void http_plugin::set_program_options(options_description&, options_description& cfg) {
-      cfg.add_options()
-            ("http-server-address", bpo::value<string>()->default_value("127.0.0.1:8888"),
-             "The local IP and port to listen for incoming http connections; set blank to disable.")
+      my->mangle_option_names();
+      if(current_http_plugin_defaults.default_unix_socket_path.length())
+         cfg.add_options()
+            (my->unix_socket_path_option_name.c_str(), bpo::value<boost::filesystem::path>()->default_value(current_http_plugin_defaults.default_unix_socket_path),
```

Comment:
Needs to be std::string with manual parsing into bfs path, or paths with spaces are spuriously rejected.  (Ancient boost defect.)